modules/tinymce/src/core/main/ts/api/html/SaxParser.ts

```diff
@@ -71,6 +72,7 @@ export interface SaxParserSettings {
   pi? (name: string, text: string): void;
   start? (name: string, attrs: AttrList, empty: boolean): void;
   text? (text: string, raw?: boolean): void;
+  dataUri? (match: Base64UriMatch): string;
```

Comment:
This feels like it's an inconsistent/bad API, as it's really not anything a streaming API should be doing since all the return types should be `void` otherwise it's not streaming (all the other callbacks are return type is void). I think this would be better suited to being passed in as a setting to `SaxParser`, something like `resolve_data_uri` maybe? or even better being pre-processed by DomParser (see my other comment)?

- [ ] Reconsider making this a callback API as it doesn't fit the current API

Reply:
Yes let's rename it to data_uri_resolver.